attribute/time.go

```diff
@@ -0,0 +1,34 @@
+package attribute
+
+import (
+	"log"
+	"time"
+
+	"github.com/getyoti/yoti-go-sdk/anchor"
+)
+
+//Time is a Yoti attribute which returns a time as its value
+type Time struct {
+	Name    string
+	Value   *time.Time
+	Type    AttrType
+	Anchors []*anchor.Anchor
+	Err     error
+}
+
+//NewTime creates a new Time attribute
+func NewTime(a *Attribute) *Time {
```

Comment:
@LKDraytonBrown (and @pn-santos?), was hoping to get your thoughts on this method - we thought about having the return type as (time, err) for this method, but this breaks up the attribute retrieval process, e.g. we can't have:
```golang
emailAddress := userProfile.EmailAddress().Value
dateOfBirth := userProfile.DateOfBirth().Value
address := userProfile.Address().Value
```
we instead need to have:
```golang
emailAddress := userProfile.EmailAddress().Value
dob, err := userProfile.DateOfBirth()
if err != nil {
    //handle error
} else {
    dateofBirth = dob
}
address := userProfile.Address().Value
```

The same is true for the StructuredPostalAddress attribute.

So we can either:
1. Return (time, err), and check for the error
1. Return time, but also have an `err` field on the struct, as is done here

I would prefer the latter, but I know this would mean that it's not obvious to clients that they need to check for this error.
I would appreciate your thoughts on whether this would be suitable from a Go perspective.

Reply:
The go dev will probably say option 1 (return time, err). Errors always are explicit to "force" the developer/client to deal with them.

Looking at the `Time` struct though, it doesn't make sense for it to have an `Err` field, error of what? An `err` is the result of some _operation_ (in go lingo, function), however that struct seems to represent only "data". Having an `err` seems to imply it represents the result of some operation but that's it's name (Time) is communicating.

Which least to the the method used: `DateOfBirth()`. This seems like an accessor method, i.e. "getting" a value/data. Why would you get an error obtaining a piece of data? If the method is actually _doing_ something (other than just returning a stored value) that can result in an error I think then the name name should reflect what it's _doing_, it should be the name of the _operation_ and not simply the name of the data in case of "success".

Reply:
This is a by product of having to use a generics alternative: 
It is indeed an accessor method, but we are having to wait until the last moment to convert the "Yoti(Attribute)" to a "Time(Attribute)", so it is not possible in the current format to retrieve it before this point. 

So you're saying it would be more appropriate to call them `GetDateOfBirth` and `GetStructuredPostalAddress` (or something similar), and return an error alongside? Or are these also accessor names that would suggest it wouldn't return an error, and it should be `ParseDateOfBirth` and `ParseStructuredPostalAddress`?

Reply:
I'm saying looking at it from a go-perspective, any function than can return an error should return `err` alongside the data it's supposed to return in a non-failure scenario.

Regarding the function names, I was trying to say that the method name should reflect what it does. If I understand you correctly you are just converting types (casting) and the error is only when that operation fails is that right? (which is unlikely to happen right?) 

If the "operation" being done is just casting from one type to the other then nvm what I said. Leaving them as-is (e.g. `DateOfBirth()`) is fine.

tl;dr I'd propose returning `(time, err)`